Add Images and Text Security Check Functionality

SnakeAmadeus opened the pull request.

SnakeAmadeus committed:
Uploading Posts now Require Security Compliance

SnakeAmadeus committed:
Optimize Detail Editing Functionality

Upon editing the page, only new images will be uploaded and old images will be deleted

zhongyizh commented on pages/post/new-post-listing/new-post-listing.js:
```diff
@@ -135,22 +145,42 @@ Page({
-		// 再把图片都uploadFile到指定文件夹，拿到图片在云上的地址
-		const { fileList } = this.data;
+		// 获取原始文件列表和当前文件列表
+		const originalFileList = this.originalCopy.fileList;    // Denote as A
```
是this.data.originalCopy.fileList

zhongyizh commented on pages/post/new-post-listing/new-post-listing.js:
```diff
@@ -135,22 +145,42 @@ Page({
-		for (let img of fileList) {
-			const fileId = await uploadImage(postId, img.url)
+		for (let img of filesToAdd) {
+			const fileId = await this.uploadImage(postId, img.url)
```
不是this，方法的位置不对

zhongyizh commented on pages/post/new-post-listing/new-post-listing.js:
```diff
@@ -135,22 +145,42 @@ Page({
+			const fileId = await this.uploadImage(postId, img.url)
 			imageUrls.push(fileId)
-			await imgSecCheck(postId, fileId)
+			let traceId = await this.imgSecCheck(postId, fileId);
```
同上

zhongyizh commented on utils/util.js:
晒生活发帖，new-post.js也要适配对应的修改

zhongyizh reviewed (changes requested):
还需要确认晒生活发帖的编辑

zhongyizh commented on pages/post/new-post-listing/new-post-listing.js:
```diff
@@ -135,22 +145,42 @@ Page({
-			mask: true
-		})
 		// 先去add Post的内容，数据库随机给一个id
 		let result = await this.uploadPostData(payload)
```
编辑贴子时不能重复调用this.uploadPostData(payload)

zhongyizh commented on pages/post/new-post-listing/new-post-listing.js:
```diff
@@ -135,22 +145,42 @@ Page({
+		const filesToAdd = currentFileList.filter(
+			img => !originalFileList.some(originalImg => originalImg.url === img.url)
+		);
+		console.log("new-post-listing: ");
```
意义不明的console.log输出

zhongyizh commented on pages/post/new-post-listing/new-post-listing.js:
```diff
@@ -35,7 +39,8 @@ Page({
         eventChannel.on('onPageEdit', (res) => {
-            this.setData(res);
+			this.setData(res);
+			this.setData({ originalCopy: res });
```
这样不行，删除图片之后this.data.originalCopy.fileList也会变
看一下handleRemove方法

zhongyizh reviewed (changes requested):
还需要控制“未通过审核”的Posts的编辑功能
审核不通过的贴子，改成只能在用户的“我的”页面自己点进去然后删除掉，同时把编辑按钮给禁用了

SnakeAmadeus committed:
Bug Fix: Edit Page Functionality now Works Properly

Warning: It is a temporary fix, it still generate a new postId upon editing the post, then it deletes the old post using this.data.originalCopy._id

Also edit button and delete button are reacting to isImgChecked property now as required.

SnakeAmadeus committed:
Clean up Delete Obsolete Imgs upon Editing

Merge it to deletePost function to reduce code

SnakeAmadeus committed:
[Under Construction] Refactor Post Creation and Editing

SnakeAmadeus committed:
Fixed the PostId Overriding Bug in Editing

Also clean up the console Informations
